vnext/Microsoft.ReactNative/RedBox.cpp

```diff
@@ -463,25 +458,114 @@ struct RedBoxHandler : public std::enable_shared_from_this<RedBoxHandler>, IRedB
       }
     }
 
-    if (auto uiQueue = m_wkUIMessageQueue.lock()) {
-      if (m_showingRedBox || !redbox) // Only show one redbox at a time
-        return;
-      m_showingRedBox = true;
-      uiQueue->runOnQueue([redboxCaptured = std::move(redbox)]() { redboxCaptured->ShowNewJSError(); });
-    }
+    if (m_showingRedBox || !redbox) // Only show one redbox at a time
+      return;
+    m_showingRedBox = true;
+
+    Mso::DispatchQueue::MainUIQueue().Post(
```

Comment:
I'm curious how Mso determines which one is the MainUIQueue - i.e. how is this going to work in xaml islands?
It might make more sense to have the react host / instance expose the dispatch queue for that island

Reply:
Yes, this is the plan. Currently we use the MainView CoreDispatcher and it fails for the XAML islands.